### PR TITLE
Let's try this once more. Poolchems, activate!

### PR DESCRIPTION
**Pull request by Kokojos**
This PR has been accepted before but needed better code, link to prior PR here : 
https://github.com/HippieStationCode/HippieStation13/pull/2711

Also fixes a minor visual bug in the display of the pool controller.

With this you can melt objects in the pool!
Make gettho cryo!
Turn everyone in the pool black with spraytan!
And much more!

### CODE REVIEW

**Comment by ghost** [comment]
Can I turn people black with spraytan and this?

**Comment by ghost** [comment]
WELL FUCK I DIDN'T READ THE DESCRIPTION
+1

**Comment by ghost** [comment]
Holy fuck you're working off of your master JFC

**Comment by Kokojos** [comment]
Yeah I merged this into my master because I had so many trouble with the map earlier din't want to take any chances. Il redownload when this passes anyway.

**Comment by ArcLumin** [comment]
+0.1 finally gonna put randomizer toxin in it

File: code/game/machinery/poolcontroller.dm [hunk 1]
```diff
@@ -57,33 +58,29 @@
...
-//herpderp				user << "A beaker is already loaded into the machine."
-//herpderp				return
-
-//herpderp			if(isrobot(user))
-//herpderp				return
+	else
+		if(stat & (NOPOWER|BROKEN))
+			return
+		if (istype(W,/obj/item/weapon/reagent_containers/glass/beaker/large))
+			if(src.beaker)
```
Comment (Carbonhell) [inline code/game/machinery/poolcontroller.dm]: src.

File: code/game/machinery/poolcontroller.dm [hunk 2]
```diff
@@ -57,33 +58,29 @@
...
-//herpderp				W.loc = src
-//herpderp				user << "You add the beaker to the machine!"
-//herpderp				updateUsrDialog()
-//herpderp				for(var/datum/reagent/R in W.reagents.reagent_list)
-//herpderp					src.cur_reagent = R.id
-//herpderp					if(adminlog)
-//herpderp						log_say("[key_name(user)] has changed the pool's chems to [R.name]")
-//herpderp						message_admins("[key_name_admin(user)] has changed the pool's chems to [R.name].")
+			if(W.reagents.total_volume >= 100 && W.reagents.reagent_list.len == 1) //check if full and allow one reageant only.
+				src.beaker =  W
```
Comment (Carbonhell) [inline code/game/machinery/poolcontroller.dm]: src.

File: code/game/machinery/poolcontroller.dm [hunk 3]
```diff
@@ -57,33 +58,29 @@
...
+					if(adminlog)
+						log_say("[key_name(user)] has changed the pool's chems to [R.name]")
+						message_admins("[key_name_admin(user)] has changed the pool's chems to [R.name].")
 
 
-//herpderp			else
-//herpderp				user << "<span class='notice'>This machine only accepts full beakers of one reagent.</span>"
-//herpderp		else
+			else
+				user << "<span class='notice'>This machine only accepts full beakers of one reagent.</span>"
```
Comment (Carbonhell) [inline code/game/machinery/poolcontroller.dm]: full LARGE beakers*

**Review by Carbonhell — CHANGES_REQUESTED** [review]

File: code/game/machinery/poolcontroller.dm (hunk 2 above)
Comment (Kokojos) [reply code/game/machinery/poolcontroller.dm]: fixed

File: code/game/machinery/poolcontroller.dm (hunk 1 above)
Comment (Kokojos) [reply code/game/machinery/poolcontroller.dm]: fixed

File: code/game/machinery/poolcontroller.dm (hunk 3 above)
Comment (Kokojos) [reply code/game/machinery/poolcontroller.dm]: fixed

**Comment by doobiedoo23** [comment]
i think i +1/3ed the old one let's do it

**Review by Carbonhell — APPROVED** [review]

**Pull request merged**